### PR TITLE
Updated scripts for on-the-fly-instrumention to with python3 and latest version of mitmproxy 

### PR DESCRIPTION
**Pull request by MadhuNimmo**

### CODE REVIEW

**Comment by msridhar** [comment]
@MadhuNimmo in the future, don't push to the `master` branch of a fork.  Create a new branch for the change, and push to the new branch and create the PR from it.

File: README.md [hunk 1]
```diff
@@ -104,17 +104,15 @@ While performing analysis in a browser, one needs to press Alt-Shift-T to end th
 **Analysis in a browser using a proxy and on-the-fly instrumentation**
 
 You can also setup a proxy to instrument JavaScript files on-the-fly.
-To do so, you need to install [mitmproxy](http://mitmproxy.org/).  We
-highly recommend **version 0.11.3**; later versions may not
-work with Jalangi.  On Linux, you can follow
+To do so, you need to install [mitmproxy](http://mitmproxy.org/). On Linux, you can follow
 [the standard installation instructions](http://docs.mitmproxy.org/en/stable/install.html),
-but instead of running `sudo pip install mitmproxy`, run `sudo pip install mitmproxy==0.11.3` to get the right version.  On Mac OS,
+but instead of running `sudo pip install mitmproxy`, run `sudo pip install mitmproxy` to get the right version.  On Mac OS,
```
Comment (msridhar) [inline README.md:109]: Can we document which version of mitmproxy we have tested with?

File: scripts/proxy.py [hunk 2]
```diff
@@ -4,54 +4,79 @@
...
 
-from distutils.version import LooseVersion
+from mitmproxy import http
+from mitmproxy import ctx
+from mitmproxy.script import concurrent
 
 from subprocess import CalledProcessError, Popen, PIPE, STDOUT
 
-p = Popen(['mitmdump --version'], stdout=PIPE, stdin=PIPE, stderr=STDOUT, shell=True)
+'''p = Popen(['mitmdump --version'], stdout=PIPE, stdin=PIPE, stderr=STDOUT, shell=True)
```
Comment (msridhar) [inline scripts/proxy.py:16]: Why are these code lines being turned into string constants?  Can we just delete them?

File: scripts/proxy.py [hunk 3]
```diff
@@ -4,54 +4,79 @@
...
 print('Jalangi home is ' + JALANGI_HOME)
 print('Current working directory is ' + WORKING_DIR)
 
+global jalangiArgs  
 jalangiArgs = ''
+global useCache 
 useCache = True
 ignore = []
 
+# For enabling/disabling instrumentation cache (enabled by default)
```
Comment (msridhar) [inline scripts/proxy.py:41]: Are these new options?  If so, maybe we should document in the README?

**Review by msridhar — CHANGES_REQUESTED** [review]
Thanks for these awesome changes!  I just have a few minor comments

File: scripts/proxy.py (hunk 3 above)
Comment (MadhuNimmo) [reply scripts/proxy.py:35]: This option was existing.

**Comment by msridhar** [comment]
@MadhuNimmo please re-request a review from me when all comments have been addressed.

**Review by msridhar — APPROVED** [review]
It would be great to have a test for this support, but we can try to do that in a follow-up

**Comment by milahu** [comment]
the readme still says that python2 is required. outdated?

https://github.com/Samsung/jalangi2/blob/fef63515086ddd14fa33078b223eddab9014889c/README.md?plain=1#L27

**Comment by msridhar** [comment]
> the readme still says that python2 is required. outdated?
> 
> https://github.com/Samsung/jalangi2/blob/fef63515086ddd14fa33078b223eddab9014889c/README.md?plain=1#L27

Yes, out of date.  Fixed in https://github.com/Samsung/jalangi2/commit/8a3669eca0486d369437eb73772f881bbdeeba7d.

**Pull request merged**